### PR TITLE
fix: [CI-4894]: Fixed condition for showing clone codebase form on Run/Retry pipeline form

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

This fixes https://harness.atlassian.net/browse/CI-4894.

✅ Added check to make sure a user sees `CI Codebase` input form only if the selected stage has `cloneCodebase` set as true(in case of a selective stage execution) or atleast one stage in the pipeline has `cloneCodebase` set as true(in case of complete pipeline execution).

✅ Moved CI module specific code out of `PipelineInputSetForm` component.

https://user-images.githubusercontent.com/48940704/180400015-8921c7b2-e5b8-4665-adce-e64239b1bc84.mov

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx [hunk 1]
```diff
@@ -272,7 +272,7 @@ function CICodebaseInputSetFormInternal({
     prCloneStrategy: prCloneStrategyOptions?.[0]?.value,
     memoryLimit: '500Mi',
     cpuLimit: '400m',
-    build: { spec: {} }
+    build: RUNTIME_INPUT_VALUE
```
Comment (vardanbansal-harness) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:275]: Sending an empty `spec` object in pipeline execution api call was failing due to a schema validation error, more details in this comment https://harness.atlassian.net/browse/CI-4894?focusedCommentId=332143.

Comment (mtran7) [inline src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx:275]: Not sure if this is the correct way to approach. <+input> is usually not a valid value when this form is shown

Comment (vardanbansal-harness) [reply src/modules/70-pipeline/components/PipelineInputSetForm/CICodebaseInputSetForm.tsx]: Same as above.

**Comment by mtran7** [comment]
Can take this as a follow up - when there are no runtime inputs, this should be the screen

**Review by mtran7 — APPROVED** [review]

**Pull request merged**